xGame/Java/xgame/src/main/java/com/xgame/service/ChessMatchService.java

```diff
@@ -216,4 +216,50 @@ else if(match.getBlackPlayer().getId() == playerId) {
 		var updatedMatch = matchRepo.save(match);
 		return updatedMatch.getMatchOutcome();
 	}
+	
+	/**
+	 * Denies a draw proposed by a given player's opponent. Undo opponent suggestion, match remains ongoing.
+	 * @param matchId - Id of match where a draw is being denied
+	 * @param playerId - player denying the draw
+	 * @return Status of match
+	 */
+	@Override
+	public MatchStatus denyDraw(int matchId, int playerId) {
```

Comment:
Instead of returning a MatchStatus you can just return void. If the FE doesn't get an error back, they know they deny succeeded.